File: app/src/main/java/com/example/development/sakaiclient20/networking/deserializers/AttachmentDeserializer.java
package com.example.development.sakaiclient20.networking.deserializers;

import com.example.development.sakaiclient20.persistence.entities.Attachment;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;

import java.lang.reflect.Type;

public class AttachmentDeserializer implements JsonDeserializer<Attachment> {
    @Override
    public Attachment deserialize(JsonElement raw,
                                  Type typeOfT,
                                  JsonDeserializationContext context)
            throws JsonParseException {

        JsonObject json = raw.getAsJsonObject();

        Attachment attachment = new Attachment();
        attachment.name = json.get("name").getAsString();
        attachment.url = json.get("url").getAsString();

        return attachment;
    }
}

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/networking/deserializers/AttachmentDeserializer.kt
package com.sakaimobile.development.sakaiclient20.networking.deserializers

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import com.google.gson.JsonParseException
import com.sakaimobile.development.sakaiclient20.persistence.entities.Attachment

import java.lang.reflect.Type

class AttachmentDeserializer : JsonDeserializer<Attachment> {
    @Throws(JsonParseException::class)
    override fun deserialize(raw: JsonElement,
                             typeOfT: Type,
                             context: JsonDeserializationContext): Attachment {

        val json = raw.asJsonObject

        val url = json.getStringMember("url")
        val attachment = Attachment(url!!)
        attachment.name = json.getStringMember("name")

        return attachment
    }
}
